iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/execution/operator/source/relational/aggregation/CountIfAccumulator.java

```
@@ -0,0 +1,107 @@
+import org.apache.tsfile.utils.RamUsageEstimator;
+
+import java.util.Collections;
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static org.apache.tsfile.enums.TSDataType.BOOLEAN;
+
+public class CountIfAccumulator implements TableAccumulator {
+  private static final long INSTANCE_SIZE =
+      RamUsageEstimator.shallowSizeOfInstance(CountIfAccumulator.class);
+
+  private long countState = 0;
+
+  public CountIfAccumulator(List<TSDataType> seriesDataType) {
+    checkArgument(
+        seriesDataType.size() == 1 && seriesDataType.get(0) == BOOLEAN,
+        "argument of count_if should be one boolean expression");
+  }
+
+  @Override
+  public long getEstimatedSize() {
+    return INSTANCE_SIZE;
+  }
+
+  @Override
+  public TableAccumulator copy() {
+    return new CountIfAccumulator(Collections.singletonList(BOOLEAN));
+  }
+
+  @Override
+  public void addInput(Column[] arguments) {
+    int count = arguments[0].getPositionCount();
+    for (int i = 0; i < count; i++) {
+      if (arguments[0].getBoolean(i)) {
+        countState++;
+      }
+    }
+  }
+
+  @Override
+  public void addIntermediate(Column argument) {
+    for (int i = 0; i < argument.getPositionCount(); i++) {
+      if (argument.isNull(i)) {
+        continue;
+      }
+      countState += argument.getLong(i);
+    }
+  }
+
+  @Override
+  public void evaluateIntermediate(ColumnBuilder columnBuilder) {
+    columnBuilder.writeLong(countState);
+  }
+
+  @Override
+  public void evaluateFinal(ColumnBuilder columnBuilder) {
+    columnBuilder.writeLong(countState);
+  }
+
+  @Override
+  public boolean hasFinalResult() {
+    return false;
+  }
+
+  @Override
+  public void removeInput(Column[] arguments) {
+    for (int i = 0; i < arguments[0].getPositionCount(); i++) {
+      if (arguments[0].getBoolean(i)) {
+        countState--;
+      }
+    }
+  }
+
+  @Override
+  public void addStatistics(Statistics[] statistics) {}
```

better to throw unsupported exception if we don't plan to support push down count_if into scan